pages/docs/coding-conventions/styling.en-US.mdx

```diff
@@ -0,0 +1,224 @@
+- If you want to apply styles based on the user's viewport size, use our predefined [breakpoints](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-styleguide/src/breakpoints/index.ts). For example, to apply different styles for tablet and desktop viewports, do this:
+
+  ```scss
+  // Tablet viewports
+  :global(.omrs-breakpoint-lt-desktop) {
+    .form {
+      height: calc(100vh - 9rem);
+    }
+  }
+
+  // Desktop viewports
+  :global(.omrs-breakpoint-gt-tablet) {
+    .form {
+      height: calc(100vh - 6rem);
+    }
+  }
+  ```
+
+  <Callout emoji="ℹ️" type="info">
+    Make sure to scope your styles under a class name (such as `.form` in the example above) to avoid them affecting
+    other components.
+  </Callout>
+
+- Use the [classnames](https://npm.im/classnames) library to conditionally apply styles to an element. Consider using classnames if you're interpolating multiple class names into a string. For example, the following snippet:
+
+  ```tsx
+  <NumberInput
+    allowEmpty
+    className={`${styles.textInput} ${val.className}`}
+    // other props omitted for brevity
+  />
+  ```
+
+  Could be replaced by:
+
+  ```tsx
+  import classnames from "classnames";
+
+  <NumberInput
+    allowEmpty
+    className={classNames(styles.textInput, val.className)}
+    // ... other props omitted for brevity
+  />;
+  ```
+
+  The following snippet shows a more advanced case - a `div` styled with multiple conditional styles:
+
+  ```tsx
+  return (
+    <div
+      className={`${styles.textInputContainer} ${disabled && styles.disabledInput} ${
+        !isWithinNormalRange && styles.danger
+      } ${useMuacColors ? muacColorCode : undefined}`}
+    >
+      // ... details omitted for brevity
+    </div>
+  );
+  ```
+
+  You can refactor this snippet to leverage `classnames` as follows:
+
+  ```tsx
+  import classNames from "classnames";
+
+  const containerClasses = classNames(styles.textInputContainer, {
+    [styles.disabledInput]: disabled,
+    [styles.danger]: !isWithinNormalRange,
+    [muacColorCode]: useMuacColors,
+  });
+
+  return <div className={containerClasses}>// ... details omitted for brevity</div>;
+  ```
+
+- Be wary of using newer CSS features that don't have widespread browser support. While modern CSS features can be powerful, we need to ensure our applications work across all supported browsers. Here are some guidelines:
+
+  - Check browser support on [Can I Use](https://caniuse.com/) before using newer CSS features
+  - Consider providing fallbacks for newer features:
+
+    ```scss
+    .container {
+      // Fallback for browsers that don't support gap
+      margin-right: 1rem;
+      // Modern browsers will use gap
+      gap: 1rem;
+    }
+    ```
+
+  - Be especially careful with:
+    - Container queries (limited support)
+    - CSS subgrid (partial support)
+    - `:has()` selector (no support in some browsers)
+    - CSS nesting (emerging support)
```

Comment:
Are we only supporting  latest chrome / FF / Safari? These 3 features seem to be supported according to caniuse.com.

Reply:
Technically, latest Chrome / FF. Safari is "best effort". Other browser are not explicitly supported. That said, of these features, I've only seen us use `:has()`.